proto/substrait/algebra.proto

```diff
@@ -726,6 +727,20 @@ message NestedLoopJoinRel {
   substrait.extensions.AdvancedExtension advanced_extension = 10;
 }
 
+// A mark join internally scans the left side, constructing a hash table that
```

Comment:
This claims that the hash table is built from the left side and then the right side is probed.  This is the reverse of the `HashJoinRel`.  If this intentional, then we should point this out (e.g. "Note that the table is built from the left side which is the opposite of the approach taken in the HashJoinRel)